### PR TITLE
[systeminfo] Add CPU frequency channels

### PR DESCRIPTION
**Pull request by mherwege**
Resolves https://github.com/openhab/openhab-addons/issues/15844
Resolves https://github.com/openhab/openhab-addons/issues/15608

This PR adds CPU frequency (clock speed) channels to the binding, exposing underlying functionality from the OSHI library. Note that the availability of this information may vary by platform. In my tests on Windows, I could not get reliable information. It did work well on an RPi 4.

It also solves some documentation issues.

### CODE REVIEW

**Comment by mherwege** [comment]
Build failed because of the integration tests that had minimal change. The failure was also not only in the added tests.
Is this a symptom of https://github.com/openhab/openhab-addons/issues/13524?

**Comment by J-N-K** [comment]
Why is the thing handler initialization failing?

**Comment by mherwege** [comment]
I am not entirely sure. The next line in the test (assertion on the thing handler) times out whereby the thing handler is null. That would indicate the thing handler initalization is not even running. However, running it through a debugger, the handler factory is called and the handler creation is called from there.
If you want to try, the last commit in this PR removes the update instructions again. But you could roll back that commit to test with update instructions.

**Comment by J-N-K** [comment]
Just add `.withProperties(Map.of("thingTypeVersion", "1"))` to the thing builder to prevent updates. The other solution is to get the thing from the registry and before checking it has a handler. The thing changes during the upgrade, so the "original" thing is never initialized and never gets a handler.

**Comment by mherwege** [comment]
DCO failure is because the revert commit was not signed off. If that is an issue, I can squash that with the original to solve it.

File: bundles/org.openhab.binding.systeminfo/src/main/java/org/openhab/binding/systeminfo/internal/SysteminfoBindingConstants.java [hunk 1]
```diff
@@ -278,6 +279,16 @@ public class SysteminfoBindingConstants {
      */
     public static final String CHANNEL_CPU_DESCRIPTION = "cpu#description";
 
+    /**
+     * Maximum frequency of the CPU
+     */
+    public static final String CHANNEL_CPU_MAXFREQ = "cpu#maxfreq";
+
+    /**
+     * Maximum frequency of the CPU
```
Comment (lolodomo) [inline bundles/org.openhab.binding.systeminfo/src/main/java/org/openhab/binding/systeminfo/internal/SysteminfoBindingConstants.java:288]: Please fix the description

**Review by lolodomo — CHANGES_REQUESTED** [review]
Only an ultra minor comment

File: bundles/org.openhab.binding.systeminfo/src/main/java/org/openhab/binding/systeminfo/internal/SysteminfoBindingConstants.java (hunk 1 above)
Comment (mherwege) [reply bundles/org.openhab.binding.systeminfo/src/main/java/org/openhab/binding/systeminfo/internal/SysteminfoBindingConstants.java]: done

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Comment by mherwege** [comment]
Build is now failing and I don't understand why. I rebased but it didn't help. The last commit is only a javadoc change, so cannot have an impact.

I have the impression there are other addon PR's suddenly failing the build.

I also have a failed local build.

**Comment by lolodomo** [comment]
Build was aborted after 240 minutes.
I try again.
I don't know if you have to rebase your branch or not after the Karaf changes of yesterday (my local builds are no more building today without a rebase).

**Comment by lolodomo** [comment]
Build finally succeeded.

**Comment by jlaur** [comment]
> I don't know if you have to rebase your branch or not after the Karaf changes of yesterday (my local builds are no more building today without a rebase).

This is not necessary. The PR build is performed on the PR branch merged with target branch.

**Pull request merged**